src/spec-node/singleContainer.ts

```diff
@@ -101,23 +101,25 @@ async function setupContainer(container: ContainerDetails, params: DockerResolve
 	};
 }
 
-export async function buildNamedImageAndExtend(params: DockerResolverParameters, config: DevContainerFromDockerfileConfig | DevContainerFromImageConfig) {
+export async function buildNamedImageAndExtend(params: DockerResolverParameters, config: DevContainerFromDockerfileConfig | DevContainerFromImageConfig, argImageName?: string) {
 	const imageName = 'image' in config ? config.image : getFolderImageName(params.common);
 	params.common.progress(ResolverProgress.BuildingImage);
 	if (isDockerFileConfig(config)) {
-		return await buildAndExtendImage(params, config, imageName, params.buildNoCache ?? false);
+		return await buildAndExtendImage(params, config, imageName, params.buildNoCache ?? false, argImageName);
 	}
 	// image-based dev container - extend
 	return await extendImage(params, config, imageName, 'image' in config);
 }
-async function buildAndExtendImage(buildParams: DockerResolverParameters, config: DevContainerFromDockerfileConfig, baseImageName: string, noCache: boolean) {
+
+async function buildAndExtendImage(buildParams: DockerResolverParameters, config: DevContainerFromDockerfileConfig, baseImageName: string, noCache: boolean, argImageName?: string) {
```

Comment:
We still have two image names being passed in here - we could simplify back to a single name by handling the default value in `buildNamedImageAndExtend`? (see [here](https://github.com/juzuluag/cli/pull/1))

Reply:
Maybe add both tags (when not pushing)? Is there an advantage of having the generated tag when later doing a `devcontainer up` in CI?

Reply:
I looked at the simplified version and made the changes. But either `--load` or `--push` has to be passed not both. 
This is the error if you use both: `error: push and load may not be set together at the moment`